src/Npgsql/Internal/PgTypeInfo.cs

```diff
@@ -150,7 +150,7 @@ internal bool TryBind(Field field, DataFormat format, out PgConverterInfo info)
             return true;
         case PgResolverTypeInfo resolverInfo:
             var resolution = resolverInfo.GetResolution(field);
-            if (!HasCachedInfo(resolution.Converter)
+            if (HasCachedInfo(resolution.Converter)
```

Comment:
Ha, the good old inversed logic bug...

Reply:
Yup, wasn't noticed as it was only a problem for reads over undecided type infos.

So never, except for binary export :')